File: 주다애/BruteForce/prob3.java
import java.util.*;

class Solution {
    HashSet<Integer> set = new HashSet<>();
    public int solution(String numbers) {
        int answer = 0;
        recursion("", numbers);
        // 소수 찾기
        Iterator<Integer> it = set.iterator();
        while(it.hasNext()) {
            int num = it.next();
            if(isPrime(num)) {
                answer++;
            }
        }
        return answer;
    }
    
    public void recursion(String num, String remain) {
        if(!num.equals("")) {
            set.add(Integer.valueOf(num));
        }
        if(remain.equals("")) return;
        for(int i = 0; i < remain.length(); i++) {
            recursion(num + remain.charAt(i), remain.substring(0, i) + remain.substring(i+1));
        }
    }
    
    public boolean isPrime(int num) {
        // 0, 1
        if(num == 0  || num == 1) return false;
        // 에라토스테네스 채의 limit
        int limit = (int) Math.sqrt(num);
        for(int i = 2; i <= limit; i++) {
            if(num % i == 0) return false;
        }
        return true;
    }
}

File: 전예진/BruteForce/prob3.py
from itertools import permutations

def is_prime(num):  
    num = int(num)
    
    if num == 0 or num == 1:
        return False
    
    for i in range(2, num):
        if num % i == 0:
            return False
    
    return True


def solution(numbers):
    num_lst = list(map(int, numbers))
    answer = []
    
    for i in range(1, len(num_lst)+1):
        tmp = list(permutations(num_lst, i))
        #print(tmp)
        
        for t in tmp:
            pm = ''.join(map(str, t))
 
            if is_prime(pm):
                answer.append(int(pm))
    
    
    tmp = list(set(answer))
    print(tmp)
    
    
    return len(tmp)
